dataService.js

```diff
@@ -0,0 +1,91 @@
+// Maps.
+let productMap;
+let categoryMap;
+let combinedProductMap;
+
+// Assign product ID as key and all other product information as value in map.
+function getProducts() {
+
+  // If productMap is not empty, return productMap.
+  if (productMap) {
+    return productMap;
+  } else {
+
+    // Otherwise, load the JSONs and create the map.
+    const Product = require('./src/data/products.json');
+    const productMapForReturn = new Map();
+
+    for (let food of Product.products) {
```

Comment:
While all the products in my categories.json file were food items, I understand now that this may not be true if the file is modified later. Thanks for pointing that out!